lib/common/models/Diff.rb

```diff
@@ -55,15 +56,26 @@ def self.added(local)
         self.new(ADD, nil, local)
       end
 
+      # Public: Static method that will produce a "modified" diff
+      #
+      # local - the local configuration
+      # aws - the aws resource
+      # changes - an object describing what was modified
+      def self.modified(aws, local, changes)
+        self.new(MODIFIED, aws, local, changes)
+      end
+
       # Public: Constructor
       #
       # type  - the type of the difference
       # aws   - the aws resource that's different (defaults to nil)
       # local - the local resource that's difference (defaults to nil)
-      def initialize(type, aws = nil, local = nil)
+      # changes - an object to describe what changed in a MODIFIED diff (defaults to nil)
+      def initialize(type, aws = nil, local = nil, changes = nil)
```

Comment:
If it's added or unmanaged there won't be a list of changes associated with it. Only when it's modified. I'm hoping this makes more sense when I refactor the types of Diff